src/array.cpp

```diff
@@ -77,6 +79,29 @@ namespace sparrow
         return (*this)[size() - 1];
     }
 
+    array array::slice(size_type start, size_type end) const
+    {
+        SPARROW_ASSERT_TRUE(start <= end);
+        array copy = *this;
+        arrow_proxy& arrow_proxy_copy = copy.get_arrow_proxy();
+        arrow_proxy_copy.set_offset(start);
+        arrow_proxy_copy.set_length(end - start);
+        return copy;
+    }
+
+    array array::slice_view(size_type start, size_type end) const
+    {
+        SPARROW_ASSERT_TRUE(start <= end);
+        const arrow_proxy& arrow_proxy_copy = get_arrow_proxy();
```

Comment:
why not 
```C++
auto view_proxy = get_arrow_proxy().view();
view_proxy.set_offset(start);
```

Reply:
View will use existing array and schema, it just use pointers to the c structures. In my case I have to copy the C structures to modify the offset and length withour changing the original and but I keep the pointers values of the members